src/build/build_step.go

```diff
@@ -191,9 +198,11 @@ func buildTarget(tid int, state *core.BuildState, target *core.BuildTarget) (err
 		// Note that ordering here is quite sensitive since the post-build function can modify
 		// what we would retrieve from the cache.
 		if target.PostBuildFunction != nil && !haveRunPostBuildFunction {
-			log.Debug("Checking for post-build output file for %s in cache...", target.Label)
-			if state.Cache.RetrieveExtra(target, cacheKey, target.PostBuildOutputFileName()) {
-				if postBuildOutput, err = runPostBuildFunctionIfNeeded(tid, state, target, postBuildOutput); err != nil {
+			log.Debug("Checking for post-build output for %s in cache...", target.Label)
+			if metadata := state.Cache.Retrieve(target, cacheKey); metadata != nil {
```

Comment:
Just wanted to check here - I think this creates a new scoped metadata and leaves the external one unchanged - did you mean to replace it?

Reply:
Nope it's deliberate - I don't want to overwrite the external one with nil and if it gets out of this scope it won't matter anyway (because we won't be using the cached version).